docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,966 @@
+  DPU4            N/A               -1         Online              up       N/A
+  DPU5            N/A               -1         Online              up       N/A
+  DPU6            N/A               -1         Online              up       N/A
+  DPU7            N/A               -1         Online              up       N/A
+
+
+```
+#### Pass/Fail Criteria
+ * Verify DPU offline in show chassis modules status after DPU shut down
+ * Verify DPU is shown in show chassis modules status after DPU powered on
+
+
+### 1.7 Check removal of pcie link between NPU and DPU
+
+#### Steps
+ * Use command `pcieutil generate` to generate pcie yaml
+ * Use `show platform pcieinfo -c` to run the pcie info test to check everything is passing
+ * Use command `echo 1 > /sys/bus/pci/devices/BUS_ID/remove` to remove pcie link between NPU and one DPU
+ * Use `show platform pcieinfo -c` to run the pcie info test to check pcie link has been removed
+ * Use command `echo 1 > /sys/bus/pci/rescan` to rescan pcie links
+ * Use `show platform pcieinfo -c` to run the pcie info test to check everything is passing
+ * This test is to check the PCie hot plug functinality since there is no OIR possible
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch: Showing example of one DPU pcie link
+
+root@sonic:/home/cisco# pcieutil generate
+Are you sure to overwrite config file pcie.yaml with current pcie device info? [y/N]: y
+Generated config file '/usr/share/sonic/device/x86_64-8102_28fh_dpu_o-r0/pcie.yaml'
+root@sonic:/home/cisco# show platform pcieinfo -c
+
+root@sonic:/home/cisco# echo 1 > /sys/bus/pci/devices/0000:1a:00.0/remove
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# echo 1 > /sys/bus/pci/rescan
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# show platform pcieinfo -c
+
+```
+#### Pass/Fail Criteria
+ * Verify after removing pcie link, pcie info test fail only for that pcie link.
+ * Verify pcieinfo test pass for all after bringing back up the link
+
+
+### 1.8 Check the NTP date and timezone between DPU and NPU
+
+#### Steps
+ * Use command `date` to get date and time zone on Switch
+ * Use command `ssh admin@169.254.x.x` to enter into required dpu.
+ * Use command `date` to get date and time zone on DPU
+   
+#### Verify in
+ * Switch and DPU
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:/home/cisco# date
+Tue 23 Apr 2024 11:46:47 PM UTC
+root@sonic:/home/cisco#
+.
+root@sonic:/home/cisco# ssh admin@169.254.24.1
+root@sonic:/home/cisco#
+.
+On DPU:
+root@sonic:/home/admin# date
+Tue 23 Apr 2024 11:46:54 PM UTC
+root@sonic:/home/cisco#
+
+```
+#### Pass/Fail Criteria
+ * Verify both the date and time zone are same
+ * Verify the syslogs on both switch and DPU to be same
+ * Verify by changing time intentionally in DPU and restart the DPU. Verify again for time sync
+
+
+### 1.9 Check the State of DPUs
+
+#### Steps
+ * Use command `show system-health DPU all` to get DPU health status. 
```

Comment:
We agreed to remove the DPU heath from the initial implementation and postpone it to a later stage. Please remove all related tests

Reply:
This DPU state is going to be there and DPU health test case is going to be in phase 2.

Added the note on test case plan itself as we discussed.

Reply:
@oleksandrivantsiv These are overall DPU health which is needed. let me know if that is not the right expectation.